### PR TITLE
More pixelpipe stuff

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
- Fixes some ints that should be gbooleans
- `dt_imgid_t` and macro usage related to `dt_act_on_get_main_image()`_

### CODE REVIEW

File: src/common/iop_profile.c [hunk 1]
```diff
@@ -1330,7 +1330,7 @@ void dt_ioppr_free_iccprofile_params_cl(dt_colorspaces_iccprofile_info_cl_t **_p
   *_dev_profile_lut = NULL;
 }
 
-int dt_ioppr_transform_image_colorspace_cl(struct dt_iop_module_t *self,
+gboolean dt_ioppr_transform_image_colorspace_cl(struct dt_iop_module_t *self,
                                            const int devid,
```
Comment (TurboGit) [inline src/common/iop_profile.c:1334]: indent please.

File: src/common/iop_profile.c [hunk 2]
```diff
@@ -1484,7 +1484,7 @@ int dt_ioppr_transform_image_colorspace_cl(struct dt_iop_module_t *self,
   return (err == CL_SUCCESS) ? TRUE : FALSE;
 }
 
-int dt_ioppr_transform_image_colorspace_rgb_cl(const int devid,
+gboolean dt_ioppr_transform_image_colorspace_rgb_cl(const int devid,
                                                cl_mem dev_img_in,
```
Comment (TurboGit) [inline src/common/iop_profile.c:1488]: likewise

File: src/common/iop_profile.h [hunk 3]
```diff
@@ -193,13 +193,13 @@ void dt_ioppr_free_iccprofile_params_cl(dt_colorspaces_iccprofile_info_cl_t **_p
                                         cl_float **_profile_lut_cl, cl_mem *_dev_profile_info,
                                         cl_mem *_dev_profile_lut);
 
-/** same as the C version */
-int dt_ioppr_transform_image_colorspace_cl(struct dt_iop_module_t *self, const int devid, cl_mem dev_img_in,
+/** same as the C version, both return TRUE in case everything went fine */
+gboolean dt_ioppr_transform_image_colorspace_cl(struct dt_iop_module_t *self, const int devid, cl_mem dev_img_in,
                                            cl_mem dev_img_out, const int width, const int height,
```
Comment (TurboGit) [inline src/common/iop_profile.h:198]: indent

File: src/common/iop_profile.h [hunk 4]
```diff
@@ -193,13 +193,13 @@ void dt_ioppr_free_iccprofile_params_cl(dt_colorspaces_iccprofile_info_cl_t **_p
...
                                            const int cst_from, const int cst_to, int *converted_cst,
                                            const dt_iop_order_iccprofile_info_t *const profile_info);
 
-int dt_ioppr_transform_image_colorspace_rgb_cl(const int devid, cl_mem dev_img_in, cl_mem dev_img_out,
+gboolean dt_ioppr_transform_image_colorspace_rgb_cl(const int devid, cl_mem dev_img_in, cl_mem dev_img_out,
                                                const int width, const int height,
```
Comment (TurboGit) [inline src/common/iop_profile.h:203]: indent

File: src/common/iop_order.c [hunk 5]
```diff
@@ -2045,11 +2045,11 @@ void dt_ioppr_insert_module_instance(struct dt_develop_t *dev, dt_iop_module_t *
   dev->iop_order_list = g_list_insert_before(dev->iop_order_list, place, entry);
 }
 
-int dt_ioppr_check_iop_order(dt_develop_t *dev,
+gboolean dt_ioppr_check_iop_order(dt_develop_t *dev,
                              const dt_imgid_t imgid,
```
Comment (TurboGit) [inline src/common/iop_order.c:2049]: indent

**Review by TurboGit — CHANGES_REQUESTED** [review]
Some minor indentation missing. TIA.

**Comment by jenshannoschwalm** [comment]
@TurboGit while "indenting" i spotted at least a possible problem.

Could you check `gboolean dt_ioppr_generate_profile_info()`, it should now correctly return the error flag.

**Review by TurboGit — APPROVED** [review]
Loos good, and indeed dt_ioppr_generate_profile_info() seems better now and return an error code instead of always 0.

**Pull request merged**